src/app/shared/services/tags/template-tag.service.ts

```diff
@@ -51,4 +52,27 @@ export class TemplateTagService implements EntityTagService {
       'tags[0].key': tag.key
     });
   }
+
+  public getAgreement(template: BaseTemplateModel, lang?: string): Observable<string> {
+    let key = this.keys.agreementDefault;
+
+    if (lang) {
+      const langKey = `${key}.${lang}`;
+      if (template.tags.find(item => item.key === langKey)) {
+        key = langKey;
+      }
+    }
+
+    return this.tagService.getList({ key, resourceId: template.id })
```

Comment:
What is this?
We need to fetch all the tags from an entity and then find the required tag in the list